dashboard/app/views/levels/_level_group.haml

```diff
@@ -27,15 +27,18 @@
 
     -# Show any text from an external level.
 
-    -# Fall back to retrieving the the last_attempt from the parsed levelgroup last_attempt
-    - sublevel_last_attempt = parsed[level.id.to_s].try(:[], 'result')
-    - if @user && current_user && @user != current_user
-      -# load other user's solution for teachers viewing their students' solution
-      - sublevel_last_attempt = @user.last_attempt(level).try(:level_source).try(:data)
-    - elsif current_user
-      -# load user's previous attempt at this puzzle.
-      - sublevel_last_attempt = current_user.last_attempt(level).try(:level_source).try(:data)
-
+    -# Show last attempted answer, unless this is a teacher looking at a student's attempt
+    -# at an anonymous survey.  In that case it's very important that we do not show the
+    -# last attempt.
```

Comment:
This logic should probably live the same place as `last_attempt` (so the `User` model, for now).

```ruby
sublevel_last_attempt = current_user.sublevel_last_attempt(level)
```

Is there a reason it can't be rolled up into `last_attempt`?

Looking at this more, it would require pulling `parsed` out of the view as well.